lib/Dialect/TTNN/Transforms/TTNNLayout.cpp

```diff
@@ -304,6 +318,11 @@ class TTNNLayoutDPSOperandsRewriter
       // TTNN Conv2d moves input, weight, and bias from host to device
       // itself. Inserting the ToLayoutOp on these operands is thus problematic.
       if (mlir::isa<ttir::Conv2dOp>(op.getOperation()) && !isResult) {
+        // For the weight input of the conv2d op, it specifically needs to be on
+        // host, so we create a host to_layout op.
+        if (operand.getOperandNumber() == 1) {
+          modified = changeLayoutToHost(op, operand, rewriter);
+        }
```

Comment:
Please open an issue to track this and tag @sdjordjevicTT - this should be moved to workarounds pass eventually.

Reply:
Added the issue here: https://github.com/tenstorrent/tt-mlir/issues/1528